.github/workflows/build_dev_documentation.yml

```diff
@@ -63,61 +88,15 @@ jobs:
           body: |
             The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_${{ env.PR_NUMBER }}). All of your documentation changes will be reflected on that endpoint.
 
-      - name: Loading cache.
-        uses: actions/cache@v2
-        id: cache
-        with:
-          path: ~/.cache/pip
-          key: v1-test_build_doc
-          restore-keys: |
-            v1-test_build_doc-${{ hashFiles('setup.py') }}
-            v1-test_build_doc
-
-      - name: Setup environment
-        run: |
-          apt-get -y update && apt-get install -y libsndfile1-dev
-
-          pip uninstall -y doc-builder
-          pip install git+https://github.com/huggingface/doc-builder -U
-          cd transformers
-          pip install .[dev]
-          cd ..
-
-          export TORCH_VERSION=$(python -c "from torch import version; print(version.__version__.split('+')[0])")
-          pip install torch-scatter -f https://data.pyg.org/whl/torch-${TORCH_VERSION}+cpu.html
-
-          pip install torchvision
-          python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'
-
-          apt install -y tesseract-ocr
-          pip install pytesseract
-          pip install pytorch-quantization --extra-index-url https://pypi.ngc.nvidia.com
-          pip install https://github.com/kpu/kenlm/archive/master.zip
-
-      - name: Setup git
-        run: |
-          git config --global user.name "Hugging Face Doc Builder"
-          git config --global user.email docs@huggingface.co
-          
-
-          cd notebooks
-          git pull origin master
-          cd ..
-          WRITE=`cat /mnt/WRITE`
-          
-          rm -rf doc-build-dev
-          git clone https://HuggingFaceDocBuilder:$WRITE@github.com/huggingface/doc-build-dev
-
       - name: Make documentation
         run: |
-          cd doc-builder &&
-          doc-builder build transformers ../transformers/docs/source --build_dir ../doc-build-dev --notebook_dir notebooks/transformers_doc --clean --version pr_$PR_NUMBER --html &&
-          cd ..
+          cd doc-build-dev && git pull && cd ..
```

Comment:
Pull right before building to reduce the risk of conflicts.

Reply:
For dev documentation, there could be `git stash && git pull && git stash apply` after the build and before the `git add .` (as the conflicts are more likely to happen with _other_ PRs which are separate folders)

And maybe this: https://mobile.twitter.com/sebastienlorber/status/1495776331287646209 (only on this action) to avoid conflicts within the same PR